core/src/test/java/dk/alexandra/fresco/framework/network/socket/ClientAuthSslSocketFactory.java

```diff
@@ -0,0 +1,59 @@
+package dk.alexandra.fresco.framework.network.socket;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.util.Objects;
+import javax.net.ServerSocketFactory;
+import javax.net.ssl.SSLServerSocket;
+import javax.net.ssl.SSLServerSocketFactory;
+
+/**
+ * Decorates a {@link SSLServerSocketFactory} so that all created server sockets are set to require
+ * client authentication.
+ */
+class ClientAuthSslSocketFactory extends ServerSocketFactory {
```

Comment:
Well, for now I put all the SSL/TLS related stuff in test, just as an example of how it can be done. At some point, maybe it should be promoted to main. But I was not sure how to do it in a nice general way that would work in main. Maybe this is an other PR on it's own.